zerver/lib/scim_filter.py

```diff
@@ -55,3 +55,23 @@ def get_extras(cls, q: str, request: HttpRequest | None = None) -> tuple[str, li
             "AND zerver_userprofile.realm_id = %s AND zerver_userprofile.is_bot = False ORDER BY zerver_userprofile.id",
             [realm.id],
         )
+
+
+class ZulipGroupFilterQuery(GroupFilterQuery):
+    attr_map = {
+        ("displayName", None, None): "zerver_namedusergroup.name",
+    }
+
+    @classmethod
+    def get_extras(cls, q: str, request: HttpRequest | None = None) -> tuple[str, list[object]]:
+        """
+        Here we ensure that results are limited to the subdomain of the request.
+        """
+        assert request is not None
+        realm = RequestNotes.get_notes(request).realm
+        assert realm is not None
+
+        return (
+            "AND zerver_namedusergroup.realm_id = %s ORDER BY zerver_namedusergroup.usergroup_ptr_id",
```

Comment:
This should likely also exclude deactivated groups. But needs some more thought, in particular settling the `DELETE` operation question in the above comment.

Reply:
this now excludes deactivated groups, as originally planned.